### ISSUE TITLE
How to understand codes below?

### ISSUE THREAD

**Issue by FinchNie** [wondergo2017/DIDA#2]
Could you elaborate on a detailed explanation? I discovered that it does not align with the content in the paper. The paper mentions, "use the variant pattern of node v at time t2 to replace the variant pattern of node u at time t1." 

```
for i in range(intervention_times):
    s1 = np.random.randint(len(sy))
    s = torch.sigmoid(sy[s1]).detach()
    conf = s * cy
    # conf=self.model.comb_pred(cs,)
    env_loss = torch.cat(
        [env_loss,
         cal_loss(conf, edge_label).unsqueeze(0)])
env_mean = env_loss.mean()
env_var = torch.var(env_loss * intervention_times)
penalty = env_mean + env_var
```

However, in the code, sy[s1] has already undergone sigmoid transformation. Why, then, during intervention, is torch.sigmoid(sy[s1]) still executed, followed by s*cy, which is peculiar?
